server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/SingleDimensionValuesSource.java

```diff
@@ -100,7 +100,7 @@
     /**
      * Sets the after value for this source. Values that compares smaller are filtered.
      */
-    abstract void setAfter(Comparable value);
+    abstract void setAfter(Comparable<?> value);
```

Comment:
This should be `Comparable<T>` IMO since this type is already generic.

Reply:
I've done as you suggest, though all the implementations seem to handle arguments other than `T`, so I'm not 100% sure it's correct.

Reply:
Hmm, I see now. Looks like we are lenient here in places and will accept string values. Given that everything compiles though that must imply we're doing either casting or reflection to call this with other types somewhere. Should be fine given type erasure but if the implementations expect other types we should allow it at compile time I suppose :shrug:

Reply:
I looked more carefully at how this method is called, and ended up reverting the wildcard replacement. It's "correct" given the implementations 🤷